### PR TITLE
Remove some seemingly unused requirements and use pip-compile

### PR DESCRIPTION
**Pull request by Twixes**
## Changes

I may totally be missing something here, but these requirements seem unused and if they're dependencies of some other requirements, then they shouldn't be in `requirements.txt`? Not sure, as pip is pretty bad at this compared to npm/yarn.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -70,19 +68,14 @@ requests-oauthlib==1.3.0
...
-sortedcontainers==2.2.2
 sqlparse==0.3.0
 tenacity==6.1.0
 toronado==0.0.11
 traitlets==4.3.3
-typing==3.7.4.3
-typing-extensions==3.7.4.2
 uritemplate==3.0.1
 urllib3==1.25.8
-uuid==1.30
```
Comment (timgl) [inline requirements.txt:85]: we definitely use uuid?

**Comment by timgl** [comment]
I think you can safely remove `uuid` and `typing` and `typing-extension`. When I pip-install and then freeze, those are the only three libraries that don't get added back in. All the other ones are just dependencies of other packages so no point in removing them

File: requirements.txt (hunk 1 above)
Comment (Twixes) [reply requirements.txt:85]: Definitely not this one!

**Comment by Twixes** [comment]
👍 BTW, we use pip-compile for dev requirements, why not prod?

**Comment by timgl** [comment]
That is a great question. We definitely could, just never got round to it. If you feel like it do go ahead

File: requirements.in [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+django-extensions==3.0.9
+django-filter==2.4.0
+django-loginas==0.3.9
+django-redis==4.12.1
+django-statsd==2.5.2
+djangorestframework==3.12.1
+djangorestframework-csv==2.1.0
+drf-exceptions-hog==0.0.4
+drf-extensions==0.6.0
+fakeredis==1.4.3
```
Comment (timgl) [inline requirements.in:35]: This is a dev requirement

**Pull request merged**